src/format_utils.h

```diff
@@ -22,13 +22,88 @@
+ * Helper function for reformatPEMCertificate:
+ * Searches for the next occurence of the given marker in pem starting at start
+ * @param pem String to search in
+ * @param marker Marker to search
+ * @param endPos Inidicates if start or end position of the marker shall be returned
+ * @param start Position to start searching at
+ * @throws MoCOCrWException if marker couldn't be found
+ * @return Start/End of next occurence of marker
+ */
+inline size_t findStartOfNextMarker(const std::string& pem, const std::string& marker, size_t start = 0)
+{
+    size_t startMarker = pem.find(marker, start);
+    if (startMarker == std::string::npos) {
+        throw MoCOCrWException("PEM seems to be invalid. Couldn't find proper begin or end markers");
+    }
+    return startMarker;
+}
+
+inline size_t findEndOfNextMarker(const std::string& pem, const std::string& marker, size_t start = 0)
+{
+    return findStartOfNextMarker(pem, marker, start) + marker.size();
+}
+
+/**
+ * Reformats a PEM encoded certificate to contain only base64 encoded lines that consist
+ * of 64 char at max. This had to be introduced to workaround a bug in OpenSSL that prevents
+ * from parsing PEM certificates that contain the base64 encoded parts in only one line and
+ * the length of the line is a multiple of 254.
+ * Issue Ticket: https://github.com/openssl/openssl/issues/9187
+ * This function only touches the base64 encoded parts between the begin- and end-marker
+ * of the certificate and doesn't touch areas before the start marker or past the end marker.
+ * However, please note that this function only reformats the first certificate found in pem.
+ *
+ * Note: to be removed as soon as the OpenSSL fix is integrated
+ *
+ * @param pem String containing the certificate to be reformatted
+ * @throws MoCOCrWException if pem doesn't contain the proper begin and end markers
+ * @returns Certificate with reformatted base64
+ */
+inline std::string reformatPEMCertificate(const std::string& pem) {
+    // search for begin-marker and extract everything from beginning of the string
+    // until the begin-markers end
+    std::string prevBase64;
+    size_t endBeginMarker = findEndOfNextMarker(pem, "-----BEGIN CERTIFICATE-----");
+    prevBase64 = pem.substr(0, endBeginMarker);
+
+    // search for end-marker and extract everything from the beginning of the marker
+    // until end of the the string
+    std::string postBase64;
+    size_t startEndMarker = findStartOfNextMarker(pem, "-----END CERTIFICATE-----", endBeginMarker);
+    postBase64 = pem.substr(startEndMarker);
+
+    // extract base64 encoded content
+    std::string base64 = pem.substr(endBeginMarker, startEndMarker - endBeginMarker - 1);
+    boost::range::remove_erase_if(base64, [](char x) { return std::isspace(x); });
+
+    // Insert newline after each 64 chars into base64 encoded content
+    std::string splittedBase64;
+    auto iter = base64.begin();
+    for (; std::distance(iter, base64.end()) >=64; iter+=64) {
```

Comment:
I don't know the actual coding style, but I guess we are adding spaces between operators and integers. If not just ignore this comment.